### PR TITLE
Devin/1732228301 initial setup

### PR DESCRIPTION
**Pull request by defiQUG**
## Summary by Sourcery

Enhance the CrossChainMessenger contract by adding a bridge fee mechanism, emergency pause/unpause functions, and a fund recovery feature. Integrate security improvements with ReentrancyGuard and Pausable. Update tests to cover new functionalities and ensure robust fee management and emergency controls.

New Features:
- Introduce a bridge fee mechanism for ETH transfers, allowing the owner to update the fee within a specified maximum limit.
- Add emergency functions to pause and unpause the contract, enhancing security and control over operations.
- Implement a fund recovery feature to allow the owner to recover stuck tokens or ETH in the contract.

Enhancements:
- Enhance the CrossChainMessenger contract by integrating ReentrancyGuard and Pausable for improved security and operational control.
- Modify the sendToPolygon function to include a bridge fee and ensure sufficient amount checks before processing transactions.

Tests:
- Add tests for the new bridge fee management, including setting, updating, and enforcing maximum fee limits.
- Implement tests for the emergency pause and unpause functionalities, ensuring only the owner can execute these actions.
- Include tests for the fund recovery feature, verifying the owner's ability to recover ETH and tokens.

### CODE REVIEW

File: contracts/CrossChainMessenger.sol [hunk 1]
```diff
@@ -6,81 +6,153 @@ import {Client} from "@chainlink/contracts-ccip/src/v0.8/ccip/libraries/Client.s
...
+        );
+        require(receiver != address(0), "Invalid receiver");
+        require(amount > 0, "Invalid amount");
 
-        emit MessageReceived(
-            any2EvmMessage.messageId,
-            receiver,
-            any2EvmMessage.destTokenAmounts.length > 0 ? any2EvmMessage.destTokenAmounts[0].amount : 0
+        // Mint WETH to receiver
+        IWETH(POLYGON_WETH).deposit{value: amount}();
```
[the remark this answers is not in the document]
Comment (BladeIV) [reply contracts/CrossChainMessenger.sol:102]: Woah Sourcery give us slow coders a chance XD
